sec1_intro_and_setup/test_app/Rakefile

```diff
@@ -0,0 +1,6 @@
+# Add your own tasks in files placed in lib/tasks ending in .rake,
```

Comment:
This file is similar to Unix Makefile which helps with building, packaging and testing Rails code. This will be used by rake utility supplied along with Ruby installation.

Reply:
Incorrect.
And, rather than say its like xxx, tell me what it is about. Don't assume I know what xxx is.

Reply:
Rake is a task runner in Ruby. It is used to handle administrative commands or tasks, which are stored either in a `Rakefile` or in a `.rake file`. One can write their own rake tasks, specific to their application. Some of the tasks are making a backup of your database, running your tests and gathering & reporting stats. Rake centralizes access to your tasks. Rake also makes a few things easier, like finding files that match a certain pattern & that have been modified recently.

Reply:
What is a task runner?

Reply:
Rake is a task runner which enables you to define a set of tasks and the dependencies between them in a file, and then have the right thing happen when you run any given task.

Reply:
....
I ask whats a task runner then you tell me rake is a task runner? COmpletely didn't asnwer the question at all

Reply:
A task runner is a program that automatically executes processes behind the scenes. One can specify which tasks they want a task runner to perform, and it will do all the work for them. Task runner helps reduce time to perform repeated tasks such as optimizing images, minifying code, and compiling files.

Reply:
erm.
A task runner is a CLI program.
CLI is like mac os Terminal or windows command prompt.
Is like you create a C++ program without GUI.